File: day14/Java/Product.java
import java.util.Scanner;

public class Product {
    public static int recursiveProd (int num1, int num2) {
        if (num2 == 1)
            return num1;

        // If any of the numbers is zero, return 0
        if (num1 == 0 || num2 == 0)
            return 0;

        // If both numbers are less than zero negative signs can be removed
        if (num1 < 0 && num2 < 0)
            return recursiveProd (-1*num1, -1*num2);
        else if (num2 < 0)
            return recursiveProd (num2, num1);
        else
            return num1 + recursiveProd(num1, num2-1);
    }

    public static void main(String[] args) {
        Scanner input = new Scanner (System.in);
        System.out.println("/* ===== Product of numbers using recursion ===== */");

        // Take input
        System.out.print("\nEnter first number: ");
        int num1 = input.nextInt();
        System.out.print("Enter second number: ");
        int num2 = input.nextInt();

        // Print the result
        System.out.println("Product of numbers " + num1 + " and " + num2 + " is: " + recursiveProd(num1, num2));
    }
}

File: day14/Python/Product.py
"""
@author: NormalVad
@date: 25/10/2020
"""

def product(a,b):
    if(a==0 or b==0):
        return 0
    elif(b == 1):
        return a
    elif(a<0 and b<0):
        a = a*(-1)
        b = b*(-1)
        return a + product(a,b-1)
    elif(a>0 and b>0):
        return a+product(a,b-1)
    elif(a<0 and b>0):
        a = a*(-1)
        x = a+product(a,b-1)
        return x*(-1)
    else:
        b = b*(-1)
        x = a+product(a,b-1)
        return x*(-1)

print("Enter two numbers to be multiplied seperated by a space")
(a,b) = list(map(int, input().split()))
print(product(a,b))
